### ISSUE TITLE
DaskMethodsMixin compute does not work with async clients

### ISSUE THREAD

**Issue by tilign** [dask/dask#5518]
Below is a minimal repro
```
import dask.distributed as dd
import dask.bag as db

cluster = dd.LocalCluster()
client = dd.Client(cluster.scheduler_address,asynchronous=True)

bag = db.from_sequence(range(0,4))
bag.map(lambda x:x).compute()
```
Gives the error
```
/opt/conda/envs/python3/lib/python3.7/site-packages/dask/base.py in compute(*args, **kwargs)
    445     postcomputes = [x.__dask_postcompute__() for x in collections]
    446     results = schedule(dsk, keys, **kwargs)
--> 447     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    448 
    449 

TypeError: zip argument #1 must support iteration
```

It looks like compute has not been changed to deal with the coroutine when the client is asynchronous.  Presumably compute should be a coroutine in that case.

I believe this affects all versions but just in case my environment is:
```
dask                      2.3.0                      py_0    conda-forge
dask-core                 2.3.0                      py_0    conda-forge
distributed               2.5.1                      py_0    conda-forge
```

**Comment by quasiben** [comment]
@tilign thanks for the interest in the async side of dask!  A few things:

1) you can easily create a client with the following:
```python
from distributed import Client
client = Client(asynchronous=True)
```

No need to go through `LocalCluster` but if you prefer that, that's ok!

2) I think the issue is related to async interface.  It's good to remember that you can await futures directly or you can use the client interface like the following:

```python
await client.compute(bag.map(lambda x: x))
```

**Comment by tilign** [comment]
Thanks for the quick update @quasiben.  As you said:

```
await client.compute(...)
```

Works fine.  In fact I am currently getting around the issue I described above by monkey patching the compute method on the mixin:

```
    def async_aware_compute(self):
        return client.compute(self)
    dask.base.DaskMethodsMixin.compute = async_aware_compute
```

**Comment by quasiben** [comment]
@tilign great!  Ok to close ?

**Comment by tilign** [comment]
May I submit a pull request with a fix instead so that others can benefit from the fix and to avoid monkey patching?

**Comment by quasiben** [comment]
I think this is currently by design.  @jcrist, if you have time can you comment here ?

**Comment by jcrist** [comment]
Yeah, we'd need to update `compute` (and I think `persist` as well) to deal with the case when the client as asynchronous. The trick here is we can't import distributed to check, so you'll need to make whatever fix you use do a generic check. You might modify `get_scheduler` to also check the `asynchronous` attribute on the client, and have a separate code path for the async case.

**Comment by jcrist** [comment]
In general the methods weren't designed with an async context in mind. I think it makes sense to support this use case, if you'd like to submit a PR I'd happily review it.

**Comment by jakirkham** [comment]
Am curious how `persist` would be updated (given it has its own lazy evaluation behavior as well)
